File: src/web/magmac/api/Option.ts
import { Consumer } from "../../java/util/function/Consumer";
import { Function } from "../../java/util/function/Function";
import { Predicate } from "../../java/util/function/Predicate";
import { Supplier } from "../../java/util/function/Supplier";
export interface Option {mapper) : map(Function<T, R>;isPresent() : boolean;other) : orElseGet(Supplier<T>;isEmpty() : boolean;mapper) : flatMap(Function<T, Option<R>>;
	orElse(other : T) : T;predicate) : filter(Predicate<T>;other) : or(Supplier<Option<T>>;consumer) : ifPresent(Consumer<T>;
}

File: src/java/magmac/api/Option.java
package magmac.api;

import java.util.function.Consumer;
import java.util.function.Function;
import java.util.function.Predicate;
import java.util.function.Supplier;

public interface Option<T> {
    <R> Option<R> map(Function<T, R> mapper);

    boolean isPresent();

    T orElseGet(Supplier<T> other);

    boolean isEmpty();

    <R> Option<R> flatMap(Function<T, Option<R>> mapper);

    T orElse(T other);

    Option<T> filter(Predicate<T> predicate);

    Option<T> or(Supplier<Option<T>> other);

    void ifPresent(Consumer<T> consumer);
}
